src/topology/homotopy/contractible.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+
+import topology.homotopy.path
+import topology.homotopy.equiv
+
+/-!
+# Contractible spaces
+
+In this file, we define `contractible_space`, a space that is homotopy equivalent to `unit`.
+-/
+
+noncomputable theory
+
+
+namespace continuous_map
+
+/-- A map is nullhomotopic if it is homotopic to some constant map -/
+def nullhomotopic {X Y : Type*} [topological_space X] [topological_space Y] (f : C(X, Y)) : Prop :=
+∃ y : Y, homotopic f (continuous_map.const _ y)
+
```

Comment:
Can you provide some very basic API? At least that constant maps are `nullhomotopic`, and stability under composition. Thanks!